src/server/twemcache-rs/src/protocol/data/response.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2021 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+//! Responses which are sent back to a client
+
+use bytes::BytesMut;
+use segcache::Item;
+
+use super::*;
+
+pub enum MemcacheResponse {
+    Deleted,
+    End,
+    Exists,
+    Item { item: Item, cas: bool },
+    NotFound,
+    Stored,
+    NotStored,
+}
+
+impl MemcacheResponse {
+    pub fn serialize(self, buffer: &mut BytesMut) {
+        match self {
+            Self::Deleted => buffer.extend(b"DELETED\r\n"),
+            Self::End => buffer.extend(b"END\r\n"),
+            Self::Exists => buffer.extend(b"EXISTS\r\n"),
+            Self::Item { item, cas } => {
+                buffer.extend(b"VALUE ");
+                buffer.extend(item.key());
+                let f = item.optional().unwrap();
```

Comment:
Oh I guess you have to haul around a bunch of metadata. But that might be OK (e.g. create a `MemcacheMetadata` type that can be used by both request and response)

Reply:
See the upcoming push, I feel like the key/value/flags/Option<cas> are part of the protocol so doesn't hurt to say that those fields are part of an item response